skyportal/models.py

```diff
@@ -183,19 +183,78 @@ def desi_dr8_url(self):
                 f"&dec={self.dec}&size=200&layer=dr8&pixscale=0.262&bands=grz")
 
 
-GroupSource = join_model('group_sources', Group, Source)
+class Filter(Base):
+    query_string = sa.Column(sa.String, nullable=False, unique=False)
+    groups = relationship("Group", secondary="group_filters", back_populates="filters")
+
+
+GroupFilter = join_model("group_filters", Group, Filter)
+
+Candidate = join_model("candidates", Filter, Obj)
+Candidate.created_at = sa.Column(sa.DateTime, nullable=False, default=datetime.now,
```

Comment:
@dannygoldstein your diagram had both `created_at` and `passed_at` columns, but it seems to me that those would be the same (we create the row when it passes a filter), so I left off `passed_at` -- please let me know if you anticipate situations where we'd need a separate `passed_at` that differs from `created_at`

Reply:
Added `passed_at` below as a nullable column